test/unit/subnet/ipv6_test.rb

```diff
@@ -0,0 +1,149 @@
+    @subnet.network = nil
+    refute @subnet.valid?
+
+    @subnet.network = old_network
+    assert @subnet.valid?
+  end
+
+  test "should have a mask" do
+    @subnet = FactoryGirl.build(:subnet_ipv6)
+    old_mask = @subnet.mask
+    @subnet.mask = nil
+    refute @subnet.valid?
+
+    @subnet.mask = old_mask
+    assert @subnet.valid?
+  end
+
+  test "cidr setter should set the mask" do
+    @subnet = FactoryGirl.build(:subnet_ipv6)
+    @subnet.cidr = 64
+    assert_equal 'ffff:ffff:ffff:ffff::', @subnet.mask
+  end
+
+  test "network should have ip format" do
+    @subnet = FactoryGirl.build(:subnet_ipv6, :network => "asf:fwe6::we6s:q1")
+    refute @subnet.valid?
+  end
+
+  test "mask should have ip format" do
+    @subnet = FactoryGirl.build(:subnet_ipv6, :mask => "asf:fwe6::we6s:q1")
+    refute @subnet.valid?
+  end
+
+  test "network should be unique" do
+    first = FactoryGirl.create(:subnet_ipv6)
+    subnet = FactoryGirl.build(:subnet_ipv6, :network => first.network)
+    refute subnet.valid?
+  end
+
+  test "network should be unique, after normalization" do
+    FactoryGirl.create(:subnet_ipv6, :network => '2001:db8::')
+    subnet = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8:0000::')
+    refute subnet.valid?
+  end
+
+  test "the name should be unique in the domain scope" do
+    first = FactoryGirl.create(:subnet_ipv6, :with_domains)
+    subnet = FactoryGirl.build(:subnet_ipv6, :name => first.name, :domains => first.domains)
+    refute subnet.valid?
+  end
+
+  test "when to_label is applied should show the domain, the mask and network" do
+    subnet = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8::', :name => 'subnet')
+    assert_equal "subnet (2001:db8::/64)", subnet.to_label
+  end
+
+  test "should find the subnet by ip" do
+    subnet = FactoryGirl.create(:subnet_ipv6)
+    assert_equal subnet, Subnet::Ipv6.subnet_for(get_ip(subnet, 10))
+  end
+
+  test "from cant be bigger than to range" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.to = get_ip(s, 10)
+    s.from = get_ip(s, 17)
+    refute s.valid?
+  end
+
+  test "should be able to save ranges" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.from = get_ip(s, 10)
+    s.to = get_ip(s, 17)
+    assert s.save
+  end
+
+  test "should not be able to save ranges if they dont belong to the subnet" do
+    s = FactoryGirl.build(:subnet_ipv6, :network => '2001:db8:1::')
+    s.from = '2001:db8:2::1'
+    s.to = '2001:db8:2::2'
+    refute s.valid?
+  end
+
+  test "should not be able to save ranges if one of them is missing" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.from = get_ip(s, 10)
+    refute s.valid?
+    s.to = get_ip(s, 17)
+    assert s.valid?
+  end
+
+  test "should not be able to save ranges if one of them is invalid" do
+    s = FactoryGirl.build(:subnet_ipv6)
+    s.from = get_ip(s, 10).gsub(/:[^:]*:/, ':xyz:')
+    s.to = get_ip(s, 17)
+    refute s.valid?
+  end
+
+  test "should strip whitespace before save" do
```

Comment:
If I understand this correctly, this is testing `normalize_addresses`, that's on the parent class. It's also on https://github.com/theforeman/foreman/blob/develop/test/unit/subnet/ipv4_test.rb#L149 so it's basically a duplicate test.